react/__tests__/LocationSearch.test.tsx

```diff
@@ -26,16 +61,18 @@ describe('Location Search', () => {
     expect(input).toHaveValue(
       'Praia de Botafogo, 200, Botafogo, Rio de Janeiro, RJ'
     )
+    // same as waiting loading to finish
+    expect(await screen.findByRole('button')).toBeInTheDocument()
+    input.blur()
```

Comment:
you should use `fireEvent.blur` here instead of calling the blur on the input

Reply:
Why is that? Also, did you meant `fireEvent.blur(input)`? Because it requires a param.

Reply:
yeah, I meant replace the `input.blur()` with `fireEvent.blur(input)`. You need that because RTL already handles sync of the events with the test renderer (and uses `act` behind the scenes, so those warnings you've commented are probably due to this)

Reply:
I see, will change

but fyi, the warning actually starts to appear in the "should display failure message when the search term is not found" when I use `fireEvent.blur(input)`

Reply:
nvm, that's just an intermittent warning :/. Will ignore those act warning for now